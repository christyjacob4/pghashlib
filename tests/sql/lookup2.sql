-- Test basic hash functionality with text
SELECT lookup2('hello world');

-- Test hash with different text inputs
SELECT lookup2('test string');
SELECT lookup2('another test');

-- Test text input with custom initval
SELECT lookup2('hello world', 42);
SELECT lookup2('hello world', 123);

-- Test bytea input
SELECT lookup2('hello world'::bytea);

-- Test bytea input with custom initval
SELECT lookup2('hello world'::bytea, 42);

-- Test integer input
SELECT lookup2(12345);
SELECT lookup2(-12345);

-- Test integer input with custom initval
SELECT lookup2(12345, 42);
SELECT lookup2(-12345, 123);

-- Test consistency (same input should give same hash)
SELECT lookup2('consistent test') = lookup2('consistent test');

-- Test initval effect (same input, different initvals should give different hashes)
SELECT lookup2('initval test', 1) != lookup2('initval test', 2);

-- Test function properties
SELECT 
    proname,
    provolatile,
    proisstrict
FROM pg_proc 
WHERE proname = 'lookup2'
ORDER BY proname, proargtypes;

-- Test extension metadata
SELECT 
    extname,
    extversion
FROM pg_extension 
WHERE extname = 'hashlib';