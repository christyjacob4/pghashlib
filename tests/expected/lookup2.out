-- Test basic hash functionality with text
SELECT lookup2('hello world');
  lookup2  
-----------
 447289830
(1 row)

-- Test hash with different text inputs
SELECT lookup2('test string');
  lookup2   
------------
 1538088394
(1 row)

SELECT lookup2('another test');
  lookup2   
------------
 1074141746
(1 row)

-- Test text input with custom initval
SELECT lookup2('hello world', 42);
   lookup2   
-------------
 -1287150784
(1 row)

SELECT lookup2('hello world', 123);
   lookup2   
-------------
 -1013821216
(1 row)

-- Test bytea input
SELECT lookup2('hello world'::bytea);
  lookup2  
-----------
 447289830
(1 row)

-- Test bytea input with custom initval
SELECT lookup2('hello world'::bytea, 42);
   lookup2   
-------------
 -1287150784
(1 row)

-- Test integer input
SELECT lookup2(12345);
   lookup2   
-------------
 -1902978736
(1 row)

SELECT lookup2(-12345);
  lookup2   
------------
 1886942764
(1 row)

-- Test integer input with custom initval
SELECT lookup2(12345, 42);
  lookup2   
------------
 1668431274
(1 row)

SELECT lookup2(-12345, 123);
  lookup2   
------------
 2032930975
(1 row)

-- Test consistency (same input should give same hash)
SELECT lookup2('consistent test') = lookup2('consistent test');
 ?column? 
----------
 t
(1 row)

-- Test initval effect (same input, different initvals should give different hashes)
SELECT lookup2('initval test', 1) != lookup2('initval test', 2);
 ?column? 
----------
 t
(1 row)

-- Test function properties
SELECT 
    proname,
    provolatile,
    proisstrict
FROM pg_proc 
WHERE proname = 'lookup2'
ORDER BY proname, proargtypes;
 proname | provolatile | proisstrict 
---------+-------------+-------------
 lookup2 | i           | t
 lookup2 | i           | t
 lookup2 | i           | t
 lookup2 | i           | t
 lookup2 | i           | t
 lookup2 | i           | t
(6 rows)

-- Test extension metadata
SELECT 
    extname,
    extversion
FROM pg_extension 
WHERE extname = 'hashlib';
 extname | extversion 
---------+------------
 hashlib | 0.0.1
(1 row)